### PR TITLE
fix(statics): use `utxolibName` instead of redefining constants

### PR DESCRIPTION
**Pull request by OttoAllmendinger**
The network constants are already defined in utxolib, there is no need
to repeat them in `statics`.

Also add `BitcoinGoldTestnet`.

Sanity check:

```
» yarn ts-node -e '
import * as utxolib from "@bitgo/utxo-lib";
import { coins, UtxoCoin } from "./src";

for (const name in utxolib.networks) {
  console.log(
    name,
    coins
      .filter((coin => (coin as UtxoCoin).network.utxolibName === name))
      .map(coin => coin.name)
  );
}
'

bitcoin [ 'btc' ]
testnet [ 'tbtc' ]
bitcoincash [ 'bch', 'bcha' ]
bitcoincashTestnet [ 'tbch', 'tbcha' ]
bitcoingold [ 'btg' ]
bitcoingoldTestnet [ 'tbtg' ]
bitcoinsv [ 'bsv' ]
bitcoinsvTestnet [ 'tbsv' ]
dash [ 'dash' ]
dashTest [ 'tdash' ]
litecoin [ 'ltc' ]
litecoinTest [ 'tltc' ]
zcash [ 'zec' ]
zcashTest [ 'tzec' ]
```

Issue: BG-39753

### CODE REVIEW

File: modules/statics/src/networks.ts [hunk 1]
```diff
@@ -158,90 +94,87 @@ class AvalancheCTestnet extends Testnet implements AccountNetwork {
...
-  messagePrefix = '\x18Bitcoin Gold Signed Message:\n';
-  bech32 = 'btg';
-  pubKeyHash = 0x26;
-  scriptHash = 0x17;
   family = CoinFamily.BTG;
+  utxolibName = 'bitcoingold';
+  explorerUrl = 'https://btgexplorer.com/tx/';
+}
+
+class BitcoinGoldTestnet extends Mainnet implements UtxoNetwork {
```
Comment (sangaman) [inline modules/statics/src/networks.ts:160]: Should this extend `Testnet`?

**Review by sangaman — CHANGES_REQUESTED** [review]
Is this a `fix` or a `refactor`? Curious if this is causing any problems currently. Either way it's good to remove redundant code. I just have one note below.

File: modules/statics/src/networks.ts (hunk 1 above)
Comment (OttoAllmendinger) [reply modules/statics/src/networks.ts:160]: yes!

**Comment by OttoAllmendinger** [comment]
> Is this a `fix` or a `refactor`? Curious if this is causing any problems currently. Either way it's good to remove redundant code. I just have one note below.

Kind of: I can't really use `@bitgo/statics` in BitGoJS at the moment because we need the objects defined by `utxolib.networks` there (not structural equality but referential equality)

**Review by sangaman — APPROVED** [review]

**Pull request merged**